src/psyclone/parse/algorithm.py

```diff
@@ -926,6 +1045,7 @@ def __init__(self, form, text, varname=None):
                 "algorithm.py:Alg:__init__: Unknown arg type provided. "
                 "Expected one of {0} but found "
                 "'{1}'.".format(str(Arg.form_options), form))
+        self._datatype = datatype
```

Comment:
Please add a comment here and document the `datatype` argument in the class docstring.

Reply:
Oops, forgot the docstring.